### PR TITLE
New features

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -56,8 +57,14 @@ jar {
     }
 }
 
+configurations.configureEach() {
+    resolutionStrategy {
+        force "org.yaml:snakeyaml:$snakeyaml_version"
+    }
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle]: Looks like it isn't required now

File: build.gradle [hunk 2]
```diff
@@ -56,8 +57,14 @@ jar {
...
+}
+
 dependencies {
-    api platform('com.exactpro.th2:bom:4.2.0')
+    api platform('com.exactpro.th2:bom:4.3.0')
```
Comment (Nikita-Smirnov-Exactpro) [inline build.gradle]: Try to migrate to 4.4.0 version

File: src/main/kotlin/com/exactpro/th2/pico/operator/config/fields/PrometheusPortsConfig.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022-2022 Exactpro (Exactpro Systems Limited)
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/pico/operator/config/fields/PrometheusPortsConfig.kt]:
```suggestion
 * Copyright 2023 Exactpro (Exactpro Systems Limited)
```

File: src/main/kotlin/com/exactpro/th2/pico/operator/generator/impl/MqConfigHandler.kt [hunk 4]
```diff
@@ -40,7 +40,8 @@ class MqConfigHandler(private val resource: BoxResource) : ConfigHandler() {
             ResourceType.Th2Estore to MqRouterConfigFactoryEstore(schemaName),
             ResourceType.Th2Mstore to MqRouterConfigFactoryMstore(schemaName),
             ResourceType.Th2Box to MqRouterConfigFactoryBox(schemaName),
-            ResourceType.Th2CoreBox to MqRouterConfigFactoryBox(schemaName)
+            ResourceType.Th2CoreBox to MqRouterConfigFactoryBox(schemaName),
+            ResourceType.Th2Job to MqRouterConfigFactoryBox(schemaName)
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/pico/operator/generator/impl/MqConfigHandler.kt:44]: How could we run jobs in pico?

File: src/main/kotlin/com/exactpro/th2/pico/operator/generator/impl/PrometheusConfigHandler.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2022-2022 Exactpro (Exactpro Systems Limited)
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/pico/operator/generator/impl/PrometheusConfigHandler.kt]:
```suggestion
 * Copyright 2023 Exactpro (Exactpro Systems Limited)
```

File: src/main/kotlin/com/exactpro/th2/pico/operator/mq/factory/MqRouterConfigFactory.kt [hunk 6]
```diff
@@ -37,4 +37,45 @@ interface MqRouterConfigFactory {
...
     )
+
+    fun generateDeclaredQueues(
+        resource: BoxResource
+    ): MutableMap<String, QueueConfiguration> {
+        val queues: MutableMap<String, QueueConfiguration> = HashMap()
+        val boxName = resource.metadata.name
+
+        // add configurations for the rest of the pins
+        for ((pinName, attributes, filters) in resource.spec.pins?.mq?.publishers ?: ArrayList()) {
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/pico/operator/mq/factory/MqRouterConfigFactory.kt]: Maybe use emptyList instead of ArrayList()

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

File: src/main/kotlin/com/exactpro/th2/pico/operator/generator/impl/MqConfigHandler.kt (hunk 4 above)
Comment (ghost) [reply src/main/kotlin/com/exactpro/th2/pico/operator/generator/impl/MqConfigHandler.kt:44]: This part of pico is just a config generation. actually running the job would be the responsibility of pico runner app which is a separate project. The config generator will just generate all needed configs similar to the regular th2 box.

**Comment by Nikita-Smirnov-Exactpro** [comment]
The dev-version-1 branch has been made from this branch

**Pull request closed without merging**